docs/security/authorization/spark/install.md

```diff
@@ -74,6 +74,25 @@ for pointing to the right Ranger admin server.
 
 </configuration>
 ```
+- Additional configs for using Marcos in row-level filter
+  - Macros are now supported for using user/group/tag in row-filter condition expression, introduced in [Ranger 2.3](https://cwiki.apache.org/confluence/display/RANGER/Apache+Ranger+2.3.0+-+Release+Notes). This feature helps significantly simplify the row-filter condition expression by replacing explicit condition query by using user/group/tag's attributes. 
+  - Consider user `bowenliang123` with an attribute `born_city` of value `Guangzhou `, the row filter condition as `city='${{USER.born_city}}'` will be transformed to `city='Guangzhou'` in execution plan.
+  - More supported marcros and usage refer to [RANGER-3605](https://issues.apache.org/jira/browse/RANGER-3605) and [RANGER-3550](https://issues.apache.org/jira/browse/RANGER-3550).
+  - Add the follow configs to `ranger-spark-security.xml` to enable UserStore Enricher which is required by macros and scripts.
+   
+```xml
+    <property>
+        <name>ranger.plugin.spark.enable.implicit.userstore.enricher</name>
+        <value>true</value>
+        <description>Enable UserStoreEnricher for fetching user and group attributes if using macros or scripts in row-filters since Ranger 2.3</description>
+    </property>
+
+    <property>
+        <name>ranger.plugin.hive.policy.cache.dir</name>
```

Comment:
ranger.plugin.hive?

Reply:
Yes.

Cache dir is required in ` org.apache.ranger.plugin.contextenricher.RangerUserStoreEnricher.java ` （https://github.com/apache/ranger/blob/master/agents-common/src/main/java/org/apache/ranger/plugin/contextenricher/RangerUserStoreEnricher.java#L84)

` String cacheDir      = getConfig(propertyPrefix + ".policy.cache.dir", null);`
and the propertyPrefix is from serviceDef name 
`String propertyPrefix    = "ranger.plugin." + serviceDef.getName();`

and the serviceDef name is depending on the specific ranger policy, which in previous config example and my case is hive.

Reply:
a plugin policy cache key often likes `ranger.plugin.spark.policy.cache.dir` where 'spark' is our plugin name,

now we have ranger.plugin.hive.policy.cache.dir and ranger.plugin.spark.policy.cache.dir
in the same config file, this is truly ambiguous.

Reply:
😭 But the macros just won't work if we skip configuring `ranger.plugin.hive.policy.cache.dir`. All these annoying implicit configs is my initial idea of submit this issue/PR to inform other users interested in Ranger macro feature and save their time.

From practical view, I think we might have to keep this.

Reply:
updated docs and clearified difference between ranger.plugin.hive.policy.cache.dir and ranger.plugin.spark.policy.cache.dir configs.